### PR TITLE
configure eslint rules to work with ts

### PR DESCRIPTION
**Pull request by thekiiingbob**
Enabled ES Lint for TS files
Fixed linting errors

### CODE REVIEW

File: cypress/tsconfig.json [hunk 1]
```diff
@@ -4,9 +4,7 @@
     "baseUrl": "../node_modules",
     "target": "es5",
     "lib": ["es5", "dom"],
-    "types": ["cypress"],
+    "types": ["cypress"]
   },
-  "include": [
-    "**/*.ts"
-  ]
+  "include": ["**/*.ts", "integration/sample.spec.js"]
```
Comment (omerose) [inline cypress/tsconfig.json]: did you intend to put sample.spec.js in here?

**Review by omerose — CHANGES_REQUESTED** [review]
is this an intended change?

File: cypress/tsconfig.json (hunk 1 above)
Comment (thekiiingbob) [reply cypress/tsconfig.json]: ah, no :) Will remove.

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -1,3 +1,16 @@
+
+// Must be declared global to be detected by typescript (allows import/export)
```
Comment (omerose) [inline cypress/support/commands.ts:2]: why did this move to the top of the file?

**Review by omerose — CHANGES_REQUESTED** [review]

File: cypress/support/commands.ts (hunk 2 above)
Comment (thekiiingbob) [reply cypress/support/commands.ts:2]: There was a linting error saying Cypress was used before the type was declared.

**Review by omerose — APPROVED** [review]
works like a charm. Although, I like to PIN dependencies since a lot of libraries do not use SEMVER conventions. I'll make a separate PR addressing that.

Cheers!

**Pull request merged**